src/java/org/apache/cassandra/service/DataResolver.java

```diff
@@ -248,13 +256,22 @@ private PartitionIterator resolveInternal(ResolveContext context,
                : Transformation.apply(counted, new EmptyPartitionsDiscarder());
     }
 
-    private class RepairMergeListener implements UnfilteredPartitionIterators.MergeListener
+    @VisibleForTesting
+    class RepairMergeListener implements UnfilteredPartitionIterators.MergeListener
     {
+        @VisibleForTesting
+        final MutableInt finishedRepairs = new MutableInt();
```

Comment:
This stuff is propagated there through constructor, I was not able to make this class static because then it broke other things.